### ISSUE TITLE
Installation failes due wrong dependencies

### ISSUE THREAD

**Issue by afle** [ioBroker/ioBroker.zigbee#503]
Running on Raspi 4:

pi@iobroker:/opt/iobroker $ sudo ./iobroker add zigbee  --host iobroker
NPM version: 6.13.4
npm install iobroker.zigbee@0.10.2 --loglevel error --prefix "/opt/iobroker" (System call)
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for concentrate@0.2.3: wanted: {"node":"~0.10.0"} (current: {"node":"10.18.1","npm":"6.13.4"})
npm ERR! notsup Not compatible with your version of node/npm: concentrate@0.2.3
npm ERR! notsup Not compatible with your version of node/npm: concentrate@0.2.3
npm ERR! notsup Required: {"node":"~0.10.0"}
npm ERR! notsup Actual:   {"npm":"6.13.4","node":"10.18.1"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/iobroker/.npm/_logs/2020-01-27T22_53_13_342Z-debug.log
host.iobroker Cannot install iobroker.zigbee@0.10.2: 1

**Comment by thatso** [comment]
You are not alone: the same error shows up on x64 with Debian 10.

**Comment by kirovilya** [comment]
Switch to Latest repository and install iobroker.zigbee@0.10.2 version

**Comment by thatso** [comment]
No change:

$ ./iobroker upgrade zigbee
Update zigbee from @0.11.5 to @1.0.1
host.iobroker Adapter "system.adapter.zigbee.0" is stopped.
NPM version: 6.13.4
npm install iobroker.zigbee@1.0.1 --loglevel error --prefix "/opt/iobroker" (System call)
npm
 ERR! code ENOTSUPnpm ERR! notsup Unsupported engine for dissolve@0.3.3: **wanted: {"node":"~0.10.0"} (current: {"node":"10.18.1"**,"npm":"6.13.4"})npm ERR! notsup Not compatible with your version of node/npm: dissolve@0.3.3npm ERR! notsup Not compatible with your version of node/npm: dissolve@0.3.3npm ERR! notsup Required: {"node":"~0.10.0"}npm ERR! notsup Actual:   {"npm":"6.13.4","node":"10.18.1"}

npm ERR! A complete log of this run can be found in:npm ERR!     /home/iobroker/.npm/_logs/2020-01-28T17_04_50_513Z-debug.log

There seems to by a typo in the wanted version?

**Comment by afle** [comment]
$ ./iobroker url "https://github.com/ioBroker/ioBroker.zigbee/tarball/master" zigbee --debug worked for me

The adapter is installed. BUT there was no zigbee instance created... (?)

**Comment by afle** [comment]
Instance was created now.

**Comment by thatso** [comment]
Same for me. I've changed the active repository in main settings to _latest_ and could install 1.0.1 ... however, I had to manually remove all zigbee* folders in `/opt/iobroker/node_modules/zigbee*` as neither update nor uninstall of the existing version would work.
Now that I think of it, I suspect uninstall failed because I did not remove all running adapter instances (actually only one) beforehand?

**Comment by mickym2** [comment]
No - I had also some problems with the installation of 1.0.1 - but adding an additional instance fixed that for me - as then missing files seems to be added as well.
But  Id didn't remove any objects manually - and after the installation was successful - all existing objects worked properly without any modification. So I cannot confirm that deinstallation or removal of objects is necessary for the update. There is in my opinion a bug in the installation script, but when the adapter fixed that the new version works properly and much more stable. I can even see no all router connections properly in the map - and in the device overview have battery (if available) and link quality available. So I think a big step forward - but the installation procedure has some problems.

**Comment by Horch79** [comment]
Hello, I want to use ZigBee in iobroker, but cannot install this module. I don't know how to install an earlier version. I think the problem is in the version number of the ZigBee module, it has to be 1.0.2, instead of 0.10.2 do I see that correctly?

**Comment by mickym2** [comment]
1.0.2 is the latest - and I installed it. But this installation process ended as with the version 1.0.1 with a lot of installation errors and instance didn't start. I created an additional instance, which I deleted afterwards. The behavior is always the same. The adapter recognizes which modules are missing and the after 10-20 minutes everything works fine.
version 0.10.2 is the latest - if you select the default repository. so you have to switch the repository to latest, to get version 1.0.2
To install a specific version you have to switch to expert mode in adapters view - but I would not recommend that.

**Comment by Vaskes1986** [comment]
Hello.
So how to update version of zigbee adapter, could somebody put here exact commans. Because I've got same error and version 0.10.2 zigbee
npm ERR! notsup Required: {"node":"~0.10.0"}
npm ERR! notsup Actual: {"npm":"6.13.4","node":"10.18.1"}
I've tried too many commands:
iobroker url "https://github.com/ioBroker/ioBroker.zigbee/tarball/master" zigbee --debug
npm i iobroker.zigbee
iobroker upload
with and without sudo

**Comment by mickym2** [comment]
So in other thread in the iobroker forum - we found out - that you have to install version 1.0.0 before you can upgrade to the current version.
So what finally did - was to install a spcific version with the following command without sudo

npm i iobroker.zigbee@1.0.0

If you can read German - than you can see that this finally worked: [https://forum.iobroker.net/topic/31143/installationsproblem-mit-adapter-zigbee-1-0-3/23](url)

**Comment by Vaskes1986** [comment]
Thanks a lot. But it did not help me. I tried it, tried to install different versions, tried to install dependecies and so on. Also I tried to delete driver, but I realize that it can't be deleted using iobroker del zigbee or npm uninstall iobroker.zigbee. 
So I decided to delete it mannually: 
cd /opt/iobroker/node_modules
rm -r iobroker.zigbee
Then reboot and install zigbee driver from web interface.
Important! I did'n loose my devices, they are still connected!
Hope this story will be helpful for someone.
